pkg/agent/npl/nplagent_test.go

```diff
@@ -0,0 +1,363 @@
+// +build !windows
+
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package npl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/npl/k8s"
+	"github.com/vmware-tanzu/antrea/pkg/agent/npl/portcache"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Mock PortPortRule for tests to simulate IPTable
```

Comment:
we use GoMock for mocking interfaces for testing. Please add your interface [there](https://github.com/vmware-tanzu/antrea/blob/master/hack/update-codegen-dockerized.sh#L95) and run `make codegen`.

Reply:
Done